frontend/public/components/factory/details.tsx

```diff
@@ -63,6 +63,7 @@ export const DetailsPage = withFallback<DetailsPageProps>((props) => {
 
 export type Page = {
```

Comment:
We could reuse this type (using TS `Pick` mapped type to represent a sub-structure) in extension properties declaration.